source/lex.tex

```diff
@@ -1327,14 +1327,15 @@
 cannot be represented by any type in its list and
 an extended integer type\iref{basic.fundamental} can represent its value,
 it may have that extended integer type.
-If all of the types in the list for the \grammarterm{integer-literal}
-are signed,
-the extended integer type is signed.
-If all of the types in the list for the \grammarterm{integer-literal}
-are unsigned,
-the extended integer type is unsigned.
-If the list contains both signed and unsigned types,
-the extended integer type may be signed or unsigned.
+This extended integer type shall have the following signedness:
```

Comment:
This uses "shall" for a requirement on the implementation. We usually avoid that (except in phrases like "the implementation shall") and reserve "shall" for diagnosable requirements on a (user-written) program.